### DISCUSSION TITLE
How Argo CD handle `templates/tests` directory of a helm chart?

### DISCUSSION THREAD

**Discussion by mikutas** [argoproj/argo-cd#15302] [Q&A] [answered]
Just curious to know...
The manifests in `tests` seems not to be rendered on the server (the behavior itself is nice for me), but I can't find `--skip-tests` option is input to `helm template` while exploring code.

(But `argocd app manifests` outputs `tests` manifests)

**Comment by sneaker-xc3** [comment]
Could we get some input on this? Because its quite confusing...

**Comment by ChristianCiach** [comment, marked as answer]
The Gitops-Engine skips manifests that have a `helm.sh/hook` annotation with an unsupported value (like `test`) in this case.

See https://github.com/argoproj/gitops-engine/blob/6b2984ebc47085852a7b63a0fd0b73c52e986217/pkg/sync/ignore/ignore.go#L11

`len(hook.Types(obj)) == 0` is `true` because the `Types` function only returns supported hook types: https://github.com/argoproj/gitops-engine/blob/6b2984ebc47085852a7b63a0fd0b73c52e986217/pkg/sync/hook/helm/type.go#L38

In other words: ArgoCD doesn't explicitly skip test resources. It actually skips *all* helm hooks with unsupported helm-hook annotations; `test` just happens being one of them.

**Reply by mikutas** [reply]
Thank you for a clear answer !

**Comment by DerekTBrown** [comment]
Created an enhancement request to add support for this: https://github.com/argoproj/argo-cd/issues/19956
